src/playdate/scoreboards.nim

```diff
@@ -0,0 +1,157 @@
+    return
+
+  if score == nil:
+    callback(emptyPDScore, $errorMessage)
+    return
+    
+  let domainScore = newPDScore(value = score.value.uint32, rank = score.rank.uint32, player = $score.player)
+  playdate.scoreboards.freeScore(score)
+  callback(domainScore, $errorMessage)
+
+proc invokeAddScoreCallback(score: PDScorePtr, errorMessage: ConstChar) {.cdecl, raises: [].} =
+  let callback = privateAddScoreCallbacks.pop() # first in, first out
+  if errorMessage != nil:
+    callback(emptyPDScore, $errorMessage)
+    return
+    
+  let domainScore = newPDScore(value = score.value.uint32, rank = score.rank.uint32, player = $score.player)
+  playdate.scoreboards.freeScore(score)
+  callback(domainScore, $errorMessage)
+
+proc invokeScoresCallback(scoresList: PDScoresListPtr, errorMessage: ConstChar) {.cdecl, raises: [].} =
+  privateAccess(PlaydateScoreboards)
+  let callback = privateScoresCallbacks.pop() # first in, first out
+  if scoresList == nil and errorMessage == nil:
+    callback(emptyPDScoresList, "Playdate-nim: No scores")
+    return
+
+  if scoresList == nil:
+    callback(emptyPDScoresList, $errorMessage)
+    return
+
+  privateAccess(SDKArray)
+  let length = scoresList.count.cint
+  let cArray = SDKArray[PDScoreRaw](data: cast[ptr UncheckedArray[PDScoreRaw]](scoresList.scores), len: length)
+  var scoresSeq = newSeq[PDScore](length)
+  for i in 0 ..< length:
+    let score = cArray[i]
+    scoresSeq[i] = newPDScore(value = score.value.uint32, rank = score.rank.uint32, player = $score.player)
+  cArray.data = nil # no need for SDKArray to free the data, freeScoresList() will do it
+
+  let domainObject = newPDScoresList(boardID = $scoresList.boardID, lastUpdated = scoresList.lastUpdated, scores = scoresSeq)
+  callback(domainObject, $errorMessage)
+  playdate.scoreboards.freeScoresList(scoresList)
+
+proc invokeBoardsListCallback(boardsList: PDBoardsListPtr, errorMessage: ConstChar) {.cdecl, raises: [].} =
+  let callback = privateBoardsListCallbacks.pop() # first in, first out
+  if boardsList == nil and errorMessage == nil:
+    callback(emptyPDBoardsList, "Playdate-nim: No boards")
+    return
+
+  if boardsList == nil:
+    callback(emptyPDBoardsList, $errorMessage)
+    return
+
+  privateAccess(SDKArray)
+  let length = boardsList.count.cint
+  let cArray = SDKArray[PDBoardRaw](data: cast[ptr UncheckedArray[PDBoardRaw]](boardsList.boards), len: length)
+  var boardsSeq = newSeq[PDBoard](length)
+  for i in 0 ..< length:
+    let board = cArray[i]
+    boardsSeq[i] = newPDBoard(boardID = $board.boardID, name = $board.name)
+  cArray.data = nil # no need for SDKArray to free the data, freeBoardsList() will do it
+
+  let domainObject = newPDBoardsList(lastUpdated = boardsList.lastUpdated, boards = boardsSeq)
+
+  callback(domainObject, $errorMessage)
+  playdate.scoreboards.freeBoardsList(boardsList)
+
+proc getPersonalBest*(this: ptr PlaydateScoreboards, boardID: string, callback: PersonalBestCallback): int32 =
```

Comment:
Should we mark this as `discardable`?

Reply:
Sure thing. After some asking around I didn't figure out what the reurn value was used for or when it can be anything other than 1